### DISCUSSION TITLE
Question: Behaviour when closing "column-last"

### DISCUSSION THREAD

**Discussion by muchgitsuchwow** [YaLTeR/niri#156] [Ideas]
Hey there,

assumption: 4 columns. 3 and 4 are visible, 50% split. When I close the only window in the 4th column, only column 3 is visible on the left of the screen.

I'd think it would make sense that an implicit scroll to the left happens, so that 2 and 3 are visible.

Opinions?

**Comment by YaLTeR** [comment]
Yeah maybe that makes sense; would need to play around with it to see.

**Comment by YaLTeR** [comment]
In 8c46611c29d77f4ff92b6293966a8e3ffba20d11 I made it so that closing a window that just opened will not only move the focus back to the left, but also restore the view offset. Which, at least in some cases, fixes this case of closing the last column (for example when it is a temporary dialog).

**Comment by sullyj3** [comment]
Yeah I just wrote a FR asking for this exact behaviour, before being prompted to search and finding this.

Specifically, the way I was thinking it could work was:

When closing the rightmost column in a workspace, the remaining columns are automatically scrolled rightwards (or the view is scrolled leftwards if you prefer to think of it that way) until either
- the right edge of the new rightmost column hits the right edge of the monitor

_or_
- the left edge of the leftmost column hits the left edge of the monitor

whichever comes first.

I think it's nice to be able to instantly see as many open windows as possible without having to manually scroll back to them.

I think this behaviour would be convenient in most cases, but I'm not sure whether it's enough of a no-brainer UX improvement to be forced on everyone. I could see some people finding it jarring. Maybe it could be a setting?

**Reply by YaLTeR** [reply]
I think the current behavior works fairly well. It doesn't really break all that often, mainly when you open two temporary windows, or the app has a two-level dialog. 

Plus, it lets you keep the window on the left side if you want it (just focus back and forth to it), so you can temporarily work on something at the end of a workspace without the view constantly scrolling left and right.

**Reply by RichardFevrier** [reply]
As @sullyj3 said I would also prefer to move everything to the right edge of the monitor `if the sum of the windows width in the workspace are more or equal the width of the monitor else the left edge of the monitor`
